### PR TITLE
[freetype] Update to 2.13.2

### LINKED ISSUE #34307: [freetype] update to 2.13.2
### Library name

freetype

### New version number

2.13.2

### Other information that may be useful (release notes, etc...)

The current version in vcpkg is 2.12.1.

Release Notes 2.13.0: https://sourceforge.net/projects/freetype/files/freetype2/2.13.0/
Release Notes 2.13.1: https://sourceforge.net/projects/freetype/files/freetype2/2.13.1/
Release Notes 2.13.2: https://sourceforge.net/projects/freetype/files/freetype2/2.13.2/

### PR DESCRIPTION
**Pull request by Osyotr**
Fixes #34307

### CODE REVIEW

**Comment by dg0yt** [comment]
Of course it is possible to read the properties, but...
In the past we tried to keep the variables free from targets generator expressions etc. We don't know if they are only used for CMake. Or actually, we do know that such variables are sometimes used to generate pkg-config files or config scripts. And the code to collect the link libraries to the variables is already there. 
(Otherwise we might simply set the libraries variable to `Freetype::Freetype`.)

**Comment by dg0yt** [comment]
Related: https://gitlab.kitware.com/cmake/cmake/-/issues/25635

**Comment by Cazadorro** [comment]
@Osyotr the checks seem to have passed, is there something more that needs to be done here?

File: ports/freetype/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -2,7 +2,60 @@ cmake_policy(PUSH)
...
+_find_package(${ARGS} NAMES freetype)
+
+if(NOT TARGET Freetype::Freetype)
+    message( FATAL_ERROR "The freetype vcpkg-cmake-wrapper.cmake assumes the existence of the"
+            "'Freetype::Freetype' target, which should be exposed in freetype-config.cmake through freetype-targets.cmake" )
+endif()
+if(NOT TARGET freetype)
+    message( FATAL_ERROR "The freetype vcpkg-cmake-wrapper.cmake assumes the existence of the"
+            "'freetype' target, which should be exposed in freetype-config.cmake through freetype-targets.cmake" )
+endif()
```
Comment (dg0yt) [inline ports/freetype/vcpkg-cmake-wrapper.cmake:18]: No `FATAL_ERROR` allowed here: The wrapper is also used for `find_package(freetype 99.2 CONFIG new-component)`. When the user asks for an unsupported version or unsupported component,  it is not fatal if not `REQUIRED`. Basically, all of the following new code needs to be guarded with a suitable `if()` checking that the CMake config was found. And in the following old code, some `if()` checks may become redundant.

**Review by dg0yt — CHANGES_REQUESTED** [review]
IMO most comments could be removed.

**Comment by Osyotr** [comment]
Aligned `vcpkg-cmake-wrapper.cmake` with [CURL](https://github.com/microsoft/vcpkg/blob/81e65f71205db9fd928938c94642af4f94041104/ports/curl/vcpkg-cmake-wrapper.cmake).
These variables are set by `FindFreetype.cmake`:
```cmake
message("FREETYPE_FOUND ${FREETYPE_FOUND}")
message("FREETYPE_INCLUDE_DIRS ${FREETYPE_INCLUDE_DIRS}")
message("FREETYPE_INCLUDE_DIR_ft2build ${FREETYPE_INCLUDE_DIR_ft2build}")
message("FREETYPE_INCLUDE_DIR_freetype2 ${FREETYPE_INCLUDE_DIR_freetype2}")
message("FREETYPE_LIBRARY_DEBUG ${FREETYPE_LIBRARY_DEBUG}")
message("FREETYPE_LIBRARY_RELEASE ${FREETYPE_LIBRARY_RELEASE}")
message("FREETYPE_LIBRARY ${FREETYPE_LIBRARY}")
message("FREETYPE_LIBRARIES ${FREETYPE_LIBRARIES}")
message("FREETYPE_VERSION_STRING ${FREETYPE_VERSION_STRING}")
```

Tested all combinations of `find_package(Freetype REQUIRED MODULE)` and `find_package(Freetype REQUIRED CONFIG)`:
```
1> [CMake] FREETYPE_FOUND TRUE
1> [CMake] FREETYPE_INCLUDE_DIRS D:/vcpkg/installed/x64-windows-static/include
1> [CMake] FREETYPE_INCLUDE_DIR_ft2build D:/vcpkg/installed/x64-windows-static/include
1> [CMake] FREETYPE_INCLUDE_DIR_freetype2 D:/vcpkg/installed/x64-windows-static/include
1> [CMake] FREETYPE_LIBRARY_DEBUG D:/vcpkg/installed/x64-windows-static/debug/lib/freetyped.lib
1> [CMake] FREETYPE_LIBRARY_RELEASE D:/vcpkg/installed/x64-windows-static/lib/freetype.lib
1> [CMake] FREETYPE_LIBRARY optimized;D:/vcpkg/installed/x64-windows-static/lib/freetype.lib;debug;D:/vcpkg/installed/x64-windows-static/debug/lib/freetyped.lib
1> [CMake] FREETYPE_LIBRARIES optimized;D:/vcpkg/installed/x64-windows-static/lib/freetype.lib;debug;D:/vcpkg/installed/x64-windows-static/debug/lib/freetyped.lib;optimized;D:/vcpkg/installed/x64-windows-static/lib/zlib.lib;debug;D:/vcpkg/installed/x64-windows-static/debug/lib/zlibd.lib;optimized;D:/vcpkg/installed/x64-windows-static/lib/bz2.lib;debug;D:/vcpkg/installed/x64-windows-static/debug/lib/bz2d.lib;optimized;D:/vcpkg/installed/x64-windows-static/lib/libpng16.lib;debug;D:/vcpkg/installed/x64-windows-static/debug/lib/libpng16d.lib;optimized;D:/vcpkg/installed/x64-windows-static/lib/zlib.lib;debug;D:/vcpkg/installed/x64-windows-static/debug/lib/zlibd.lib;optimized;D:/vcpkg/installed/x64-windows-static/lib/brotlidec.lib;debug;D:/vcpkg/installed/x64-windows-static/debug/lib/brotlidec.lib;optimized;D:/vcpkg/installed/x64-windows-static/lib/brotlicommon.lib;debug;D:/vcpkg/installed/x64-windows-static/debug/lib/brotlicommon.lib
1> [CMake] FREETYPE_VERSION_STRING 2.13.2
```
Output on Linux is pretty much the same.

**Review by FrankXie05 — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**